kep/tbd-peaks-power-aware-scheduling/README.md

```diff
@@ -0,0 +1,326 @@
+  - Current node utilization can be queried from Prometheus (with the metric source provider being a service like node-exporter or load-watcher).
+  - Add the change in utilization (corresponding to the pod resource requirement) to the current utilization to get the updated utilization
+- Return the change in the node's instantaneous power:
+  - The power consumption at the current utilization can be either found from the node metrics or by referring to the power model
+  - The power consumption at the updated utilization can be found by referring to the power model
+- Normalize plugin score: The score range can be provided by the user (otherwise consider the range [0, 100] as the default)
+- Choose a suitable weight for PEAKS plugin in kube-scheduler configuration: This is specific to the environment configuration.
+  - If the K8s cluster nodes are running using non-renewable energy sources, then a high weightage for PEAKS plugin can result into reduced CO2 emissions
+
+### Pre-requisite modules
+- Prometheus
+  - For energy metrics collection
+- Load-watcher
+  - For collecting node usage metrics exported after processing metrics received from source
+- Node-exporter
+  - Source of node usage metrics
+- Kepler
+  - Source of energy consumption metrics
+
+### Use-cases (that save energy using PEAKS plugin)
+Below use-case scenarios demonstrate energy savings using PEAKS scheduler plugin over default kube-scheduler.
+- Deployment of a Pod (via kube-scheduler):
+  - On a K8s cluster with heterogeneous node configurations (resulting in the energy efficiency of the nodes not being the same), while deploying a pod, the `kube-scheduler` prefers energy efficient nodes for pod placement using PEAKS scheduler plugin vs. randomly selects a node for pod placement using default scheduler plugin.
+    - Since energy savings decrease with the increased use of energy ineffecient nodes, PEAKS plugin favours pod placement on energy efficient nodes over pod placement on energy inefficient nodes.
+  - Below graph demonstrates energy saving while kube-scheduler deploys a pod ![PEAKS with Kube-scheduler](./figs/PEAKS_with_Kube-scheduler.png)
+    - The graph in the left depicts pod placement on the less energy effecient node which was randomly selected by default kube-scheduler.
+    - The graph in the middle depicts pod placement on the more energy efficient node which was the preferred choice by PEAKS scheduler plugin.
+    - The graph in the right depicts the savings in energy consumption across the cluster nodes over time (i.e., difference in the aggregate cluster energy consumed under PEAKS plugin placement vs. default scheduler placement) increas as the pods continue execution.
+    - The energy savings will grow, when the node utilizations increase and/or pods run for longer duration (node utilizations were ~12% during this experiment which ran for 10 minutes).
+- Scaling of a Pod (via Horizontal Pod Autoscaler):
```

Comment:
I may suggest merging "scaling" and "kubecl scale" into a generic bullet as in essence the behaviors are no different - just spin up pods or destroy pods.

Reply:
Sure, we have merged as suggested.